### PR TITLE
Provide disks info during connection

### LINKED ISSUE #56: Provide SmartData and fill with it remote object

### PR DESCRIPTION
**Pull request by wkucjas**
Closes #56

### CODE REVIEW

File: src/agent/Server.cpp [hunk 1]
```diff
@@ -71,12 +72,25 @@ void Server::setOverallStatus(GeneralHealth::Health overallStatus)
     emit overallStatusChanged(m_health);
 }
 
+void Server::setDiskInfoCollection(std::vector<DiskInfo> diskInfoCollection)
```
Comment (Kicer86) [inline src/agent/Server.cpp:75]: const &

File: src/agent/Server.cpp [hunk 2]
```diff
@@ -71,12 +72,25 @@ void Server::setOverallStatus(GeneralHealth::Health overallStatus)
...
+
 
 GeneralHealth::Health Server::overallStatus() const
 {
     return m_health;
 }
 
+std::vector<DiskInfo> Server::diskInfoCollection() const
+{
+    std::vector<DiskInfo> vec;
```
Comment (Kicer86) [inline src/agent/Server.cpp]: return m_diskInfoCollection? (as const &)

File: src/agent/Server.cpp [hunk 3]
```diff
@@ -87,12 +101,30 @@ void Server::CollectInfoAboutDiscs()
...
+    auto diskCollection = diskCollector->GetDisksList();
+    std::vector< DiscStatusCalculator::ProbePtr > probes;
     auto probe = systemUtilsFactory.generalAnalyzer();
     probes.emplace_back(std::move(probe));
 
-    DiscStatusCalculator calc(probes, discCollection);
+    DiscStatusCalculator calc(probes, diskCollection);
     setOverallStatus(calc.GetStatus());
+
+    std::vector<DiskInfo> temporaryDiscInfoCollection;
```
Comment (Kicer86) [inline src/agent/Server.cpp]: why temporary?

File: src/agent/windows/CMDCommunication.cpp [hunk 4]
```diff
@@ -34,9 +48,31 @@ std::string CMDCommunication::ExecuteDiscStatusCommand(const Disk& _disk) const
...
     ret = ret.substr(statusPosStart, statusPosStop - statusPosStart);
     return ret;
-}
\ No newline at end of file
+}
+
+void CMDCommunication::ChangeStringToLowerCase(std::string& _string) const
+{
+    std::transform(_string.begin(), _string.end(), _string.begin(),
+        [](unsigned char c) { return std::tolower(c); });
```
Comment (Kicer86) [inline src/agent/windows/CMDCommunication.cpp]: wouldn't std::tolower instead of lambda work?

File: src/common/SmartDataSerialize.h [hunk 5]
```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "SmartData.h"
+#include <QDataStream>
+
+inline QDataStream& operator<<(QDataStream& _out, const SmartData::AttrData& _attrData)
```
Comment (Kicer86) [inline src/common/SmartDataSerialize.h:6]: why _ in parameter names?

**Review by Kicer86 — CHANGES_REQUESTED** [review]

File: src/agent/Server.cpp (hunk 1 above)
Comment (wkucjas) [reply src/agent/Server.cpp:75]: This is an override method from rep_AgentStatus_source.h which is generated automatically.

File: src/agent/Server.cpp (hunk 2 above)
Comment (wkucjas) [reply src/agent/Server.cpp]: good point!

File: src/agent/windows/CMDCommunication.cpp (hunk 4 above)
Comment (wkucjas) [reply src/agent/windows/CMDCommunication.cpp:77]: Would it give me any advantage?

File: src/common/SmartDataSerialize.h (hunk 5 above)
Comment (wkucjas) [reply src/common/SmartDataSerialize.h:6]: I write in this style since this project started.

File: src/agent/windows/CMDCommunication.cpp (hunk 4 above)
Comment (wkucjas) [reply src/agent/windows/CMDCommunication.cpp]: I belive we met such situation few PRs ago. And it won't.

Comment (Kicer86) [reply src/agent/windows/CMDCommunication.cpp]: something like this: std::transform(_string.begin(), _string.end(), _string.begin(), std::tolower);

**Pull request merged**